business_requirement/models/business.py

```diff
@@ -0,0 +1,444 @@
+        return states
+
+    @api.multi
+    def action_button_confirm(self):
+        self.write({'state': 'confirmed'})
+        self.confirmed_id = self.env.user
+        self.confirmation_date = fields.Datetime.now()
+
+    @api.multi
+    def action_button_back_draft(self):
+        self.write({'state': 'draft'})
+        self.confirmed_id = self.approved_id = []
+        self.confirmation_date = self.approval_date = ''
+
+    @api.multi
+    def action_button_approve(self):
+        self.write({'state': 'approved'})
+        self.approved_id = self.env.user
+        self.approval_date = fields.Datetime.now()
+
+    @api.multi
+    def action_button_customer_approval(self):
+        self.write({'state': 'customer_approval'})
+
+    @api.multi
+    def action_button_in_progress(self):
+        self.write({'state': 'in_progress'})
+
+    @api.multi
+    def action_button_done(self):
+        self.write({'state': 'done'})
+
+    @api.multi
+    def action_button_cancel(self):
+        self.write({'state': 'cancel'})
+
+    @api.multi
+    def action_button_drop(self):
+        self.write({'state': 'drop'})
+
+    @api.cr_uid_ids_context
+    def message_post(self, cr, uid, thread_id, body='', subject=None,
+                     type='notification', subtype=None, parent_id=False,
+                     attachments=None, context=None,
+                     content_subtype='html', **kwargs):
+        """
+            ---- changes start here 2016/03/08
+            Overwrite method message_post from mail.thread to modify the
+            default behavior of subject with mail messages.
+            ---- changes ends here 2016/03/08
+
+            Post a new message in an existing thread, returning the new
+            mail.message ID.
+
+            :param int thread_id: thread ID to post into, or list with one ID;
+                if False/0, mail.message model will also be set as False
+            :param str body: body of the message, usually raw HTML that will
+                be sanitized
+            :param str type: see mail_message.type field
+            :param str content_subtype:: if plaintext: convert body into html
+            :param int parent_id: handle reply to a previous message by adding
+                the parent partners to the message in case of private
+                discussion
+            :param tuple(str,str) attachments or list id: list of attachment
+                tuples in the form ``(name,content)``, where content is NOT
+                base64 encoded
+
+            Extra keyword arguments will be used as default column values
+            for the new mail.message record. Special cases:
+                - attachment_ids: supposed not attached to any document;
+                    attach them to the related document.
+                    Should only be set by Chatter.
+            :return int: ID of newly created mail.message
+        """
+        if context.get('default_model') == 'business.requirement':
```

Comment:
This is definitively a :-1: 

You can't overwrite totally this method only for making a little change. You have to find a way of getting your behavior without copy/pasting the code.